packages/yarnpkg-core/tests/LegacyMigrationResolver.test.ts

```diff
@@ -14,17 +14,17 @@ describe(`LegacyMigrationResolver`, () => {
       resolved: `https://foo.bar/@types%2fno-deps/-/no-deps-1.0.0.tgz`,
       expected: `npm:1.0.0`,
     },
-    // https://github.com/yarnpkg/berry/issues/2354
+    // https://github.com/yarnpkg/berry/issues/2354 & https://github.com/yarnpkg/berry/pull/5216
     {
       version: `1.0.0`,
       resolved: `https://company.jfrog.io/company/api/npm/registry-name/@scope/package-name/-/@scope/package-name-1.0.0.tgz#eeeec1e4e8850bed0468f938292b06cda793bf34`,
-      expected: `npm:1.0.0::__archiveUrl=https%3A%2F%2Fcompany.jfrog.io%2Fcompany%2Fapi%2Fnpm%2Fregistry-name%2F%40scope%2Fpackage-name%2F-%2F%40scope%2Fpackage-name-1.0.0.tgz%23`,
```

Comment:
That doesn't seem correct - we kept `__archiveUrl` because iirc Artifactory didn't support the "standard url patterns" (ie you can see here the tarball is inside `https://<registry>/@scope/package-name/-/@scope/package-name-1.0.0.tgz`, instead of `https://<registry>/@scope/package-name/-/package-name-1.0.0.tgz`).

Do they now support the standard url?

Reply:
I will need more info how to test this :) But for me this was problem with locally published scoped packages that were:
`@scope/package-name/-/@scope/package-name` and not
`@scope/package-name/-/package-name`

Currently on yarn@1 in `yarn.lock` we have:
`  resolved "https://artifactory.corp/artifactory/api/npm/virtual-node/@achrinza/node-ipc/-/node-ipc-9.2.6.tgz#75c89e95fe38ff9e3dd33a0d2681dc28f1c31f33"`

for package from public and for internal:

`  resolved "https://artifactory.corp/artifactory/api/npm/virtual-node/@corp/package-name/-/@corp/package-name-48.0.14.tgz#708e2351f00fb5415c90fadc8e685d743be06595"`

Other fix for me would be also to publish it under `@scope/package-name/-/package-name` but I'm not sure why `npm` does repeat `@scope`.